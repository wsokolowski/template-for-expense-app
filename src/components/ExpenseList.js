import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';
import { Link } from 'react-router-dom';

export const ExpenseList = (props) => (
    <div>
        <h1>Expense List</h1>
        { props.expenses.length === 0 ? (
            <p>No expenses</p>
        ) : (
            props.expenses.map((expense) => {
                return <ExpenseListItem key={expense.id} {...expense}/>
            })
        )}
        <Link to="/create">
            <button>Add expense</button>
        </Link>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);