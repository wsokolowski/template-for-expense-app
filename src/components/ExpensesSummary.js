import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export class ExpensesSummary extends React.Component {
    render () {
        return (
            <div>
                <p>Count: {this.props.expensesCount}</p>
                {<p>Total: {numeral(this.props.expensesTotal / 100).format('$0,0.00')}</p>}
            </div>
        );
    }
};

const mapStateToProps = (state) => {
    return {
        expensesCount: selectExpenses(state.expenses, state.filters).length,
        expensesTotal: selectExpensesTotal(selectExpenses(state.expenses, state.filters))
    };
};

export default connect(mapStateToProps)(ExpensesSummary);