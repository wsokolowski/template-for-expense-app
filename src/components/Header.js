import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="is-active" exact>Dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active" exact>Create Expense</NavLink>
        <NavLink to="/help" activeClassName="is-active" exact>Help</NavLink>
    </header>
);

export default Header;