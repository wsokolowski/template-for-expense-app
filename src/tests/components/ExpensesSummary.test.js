import React from 'react';  
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary with no expenses correctly', () =>{
    const wrapper = shallow(<ExpensesSummary expensesCount={0} expensesTotal={0}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with one expense correctly', () =>{
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={234}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with multiple expenses correctly', () =>{
    const wrapper = shallow(<ExpensesSummary expensesCount={3} expensesTotal={23456456345}/>);
    expect(wrapper).toMatchSnapshot();
});