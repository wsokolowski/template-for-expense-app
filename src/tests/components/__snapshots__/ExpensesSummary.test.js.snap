// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with multiple expenses correctly 1`] = `
<div>
  <p>
    Count: 
    3
  </p>
  <p>
    Total: 
    $234,564,563.45
  </p>
</div>
`;

exports[`should render ExpensesSummary with no expenses correctly 1`] = `
<div>
  <p>
    Count: 
    0
  </p>
  <p>
    Total: 
    $0.00
  </p>
</div>
`;

exports[`should render ExpensesSummary with one expense correctly 1`] = `
<div>
  <p>
    Count: 
    1
  </p>
  <p>
    Total: 
    $2.34
  </p>
</div>
`;
